### PR TITLE
Cryo spec one time tech purchase

### PR DESCRIPTION
**Pull request by morrowwolf**
# About the pull request

This PR puts cryo specialist in its own tech.

It can be purchased once for eight points for a single specialist. Point cost may be changed.

# Explain why it's good for the game

Moving this to its own category gives commanders a choice between a few men and a specialist while also stopping tons of spec stacking.

# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog

:cl: Morrow
add: Added cryo specialist tech that can be purchased once for eight points
/:cl:

### CODE REVIEW

File: code/datums/emergency_calls/cryo_spec.dm [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	mob_max = 1
+	mob_min = 1
+	probability = 0
+	objectives = "Assist the USCM forces"
+	name_of_spawn = /obj/effect/landmark/ert_spawns/distress_cryo
+	shuttle_id = ""
+	spawn_max_amount = TRUE
+
+/datum/emergency_call/cryo_spec/create_member(datum/mind/mind, turf/override_spawn_loc)
+	set waitfor = 0
```
Comment (harryob) [inline code/datums/emergency_calls/cryo_spec.dm:12]:
```suggestion
	set waitfor = FALSE
```

File: code/datums/emergency_calls/cryo_spec.dm [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	if(SSmapping.configs[GROUND_MAP].map_name == MAP_WHISKEY_OUTPOST)
+		name_of_spawn = /obj/effect/landmark/ert_spawns/distress_wo
+	var/turf/spawn_loc = override_spawn_loc ? override_spawn_loc : get_spawn_point()
+
+	if(!istype(spawn_loc)) return //Didn't find a useable spawn point.
```
Comment (harryob) [inline code/datums/emergency_calls/cryo_spec.dm:17]: in line control statement

File: code/modules/cm_tech/techs/marine/tier3/cryo_spec.dm [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+	flags = TREE_FLAG_MARINE
+	tier = /datum/tier/three
+
+/datum/tech/cryomarine/can_unlock(mob/M)
+	. = ..()
+	if(!.)
+		return
+	if(!SSticker.mode)
+		to_chat(M, SPAN_WARNING("You can't do this right now!"))
```
Comment (harryob) [inline code/modules/cm_tech/techs/marine/tier3/cryo_spec.dm:19]:
```suggestion
/datum/tech/cryomarine/can_unlock(mob/user)
	. = ..()
	if(!.)
		return
	if(!SSticker.mode)
		to_chat(user, SPAN_WARNING("You can't do this right now!"))
```

Comment (harryob) [reply code/modules/cm_tech/techs/marine/tier3/cryo_spec.dm:19]: also why isnt this check just on /datum/tech/proc/can_unlock i dont understand anything

File: code/datums/emergency_calls/cryo_spec.dm [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+	if(!mind)
+		for(var/obj/structure/machinery/cryopod/pod in view(7,human))
+			if(pod && !pod.occupant)
+				pod.go_in_cryopod(human, silent = TRUE)
+				break
+
+	sleep(5)
+	human.client?.prefs.copy_all_to(human, JOB_SQUAD_SPECIALIST, TRUE, TRUE)
+	arm_equipment(human, /datum/equipment_preset/uscm/spec/cryo,  mind == null, TRUE)
```
Comment (harryob) [inline code/datums/emergency_calls/cryo_spec.dm:34]: no changes but i hate how this third argument is presented

File: code/datums/emergency_calls/cryo_spec.dm [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+	else
+		human.create_hud()
+
+	if(!mind)
+		for(var/obj/structure/machinery/cryopod/pod in view(7,human))
+			if(pod && !pod.occupant)
+				pod.go_in_cryopod(human, silent = TRUE)
+				break
+
+	sleep(5)
```
Comment (harryob) [inline code/datums/emergency_calls/cryo_spec.dm:32]: i dont enjoy these sleeps but i know its just copy pasta

**Review by harryob — CHANGES_REQUESTED** [review]

**Comment by morrowwolf** [comment]
My ctrl-c, ctrl-v skills are unmatched by any mortal

**Review by harryob — APPROVED** [review]

**Pull request merged**